### PR TITLE
feat: extend Middleware trait customized for celo

### PR DESCRIPTION
**Pull request by hanyunx**
## Motivation
We would like to have a customized `provider` for Celo so as to add more functions exclusively provided by Celo blockchain.

## Solution
A new trait `CeloMiddleware` is added to extend the `Middleware` for provider, so that functions in `Middleware` will query APIs compatible with Ethereum while functions in `CeloMiddleware` will query those exclusively in Celo.

The method `get_validators_bls_public_keys()` is added as an example.

### CODE REVIEW

File: ethers-providers/src/lib.rs [hunk 1]
```diff
@@ -561,3 +561,17 @@ pub trait Middleware: Sync + Send + Debug {
             .map_err(FromErr::from)
     }
 }
+
+#[cfg(feature = "celo")]
+#[async_trait]
+pub trait CeloMiddleware: Middleware {
+    async fn get_validators_bls_public_keys(
+        &self,
```
Comment (gakonst) [inline ethers-providers/src/lib.rs:569]: This should probably be an `Option<BlockId>`? https://github.com/gakonst/ethers-rs/blob/b4f490844b5d81c40038b2c6383b0292cc8c99ed/ethers-providers/src/lib.rs#L252

File: ethers-providers/src/lib.rs [hunk 2]
```diff
@@ -561,3 +561,17 @@ pub trait Middleware: Sync + Send + Debug {
...
+        block_number: String,
+    ) -> Result<Vec<String>, ProviderError> {
```
Comment (gakonst) [inline ethers-providers/src/lib.rs:571]: How long are the keys? Is there any more specific type we could use instead of String?

**Review by gakonst — CHANGES_REQUESTED** [review]
Supportive of having a Celo middleware. Minor nits but looks good otherwise.

File: ethers-providers/src/lib.rs (hunk 2 above)
Comment (hanyunx) [reply ethers-providers/src/lib.rs:571]: The public keys are 96 bytes. Maybe we can use `Vec<u8>` instead.

**Pull request merged**